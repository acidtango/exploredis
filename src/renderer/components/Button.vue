<template>
  <button
    :tabindex="tabindex"
    :type="type"
    :disabled="disabled"
    :aria-label="label"
  >
    <slot />
  </button>
</template>

<script>
export default {
  name: 'Button',
  props: {
    /**
     * Sets the button disabled state
     */
    disabled: {
      type: Boolean,
      default: false,
    },
    /**
     * Sets the button tab index
     */
    tabindex: String,
    /**
     * Sets the button type
     */
    type: {
      type: String,
      default: 'button',
    },
  },
}
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped lang="scss" scoped>
button {
  background: #5628ee;
  padding: 10px 28px;
  color: white;
  border: 0;
  border-radius: 8px;
  font-weight: bold;
  cursor: pointer;
  transition: background 0.2s ease-in;

  &:hover {
    background: #4420be; // TODO: Use color functions?
  }

  &:disabled {
    background: #947af4; // TODO: Use color functions?
  }
}
</style>

<docs>
This button is amazing, use it responsibly.

## Examples

```jsx
<Button>Push Me</Button>
```
</docs>
