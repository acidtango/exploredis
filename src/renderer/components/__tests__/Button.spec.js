import { shallowMount } from '@vue/test-utils'
import Button from '@/components/Button.vue'

describe('<Button />', () => {
  it('renders a label', () => {
    const propsData = {
      label: 'Name',
    }
    const wrapper = shallowMount(Button, { propsData })
    expect(wrapper).toMatchSnapshot()
  })

  it('renders a placeholder', () => {
    const propsData = {
      label: 'Name',
      placeholder: 'Name',
    }
    const wrapper = shallowMount(Button, { propsData })
    expect(wrapper).toMatchSnapshot()
  })

  it('renders with disabled state', () => {
    const propsData = {
      label: 'Name',
      placeholder: 'Anakin Skywalker',
      disabled: true,
    }
    const wrapper = shallowMount(Button, { propsData })
    expect(wrapper).toMatchSnapshot()
  })

  it('renders with readOnly state', () => {
    const propsData = {
      label: 'Name',
      placeholder: 'Name',
      readOnly: true,
    }
    const wrapper = shallowMount(Button, { propsData })
    expect(wrapper).toMatchSnapshot()
  })

  it('renders with an initial value', () => {
    const propsData = {
      label: 'Name',
      placeholder: 'Name',
      value: 'General Kenobi',
    }
    const wrapper = shallowMount(Button, { propsData })
    expect(wrapper).toMatchSnapshot()
  })
})
