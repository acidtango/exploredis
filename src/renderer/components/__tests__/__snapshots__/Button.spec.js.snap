// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders a label 1`] = `<button type="button" label="Name"></button>`;

exports[`<Button /> renders a placeholder 1`] = `<button type="button" label="Name" placeholder="Name"></button>`;

exports[`<Button /> renders with an initial value 1`] = `<button type="button" label="Name" placeholder="Name" value="General Kenobi"></button>`;

exports[`<Button /> renders with disabled state 1`] = `<button type="button" disabled="disabled" label="Name" placeholder="Anakin Skywalker"></button>`;

exports[`<Button /> renders with readOnly state 1`] = `<button type="button" label="Name" placeholder="Name" readonly="true"></button>`;
